backend/migrations/versions/427db8f31999_backfill_MF_resource_permissions.py

```diff
@@ -70,12 +70,54 @@ def upgrade():
     for dataset in datasets:
         ResourcePolicyService.attach_resource_policy(
             session=session,
-            group=dataset.SamlGroupName,
+            group=dataset.SamlAdminGroupName,
             resource_uri=dataset.datasetUri,
             permissions=[ATTACH_METADATA_FORM],
             resource_type=DatasetBase.__name__,
         )
 
 
 def downgrade():
-    print('no downgrade supported')
+    bind = op.get_bind()
+    session = orm.Session(bind=bind)
+    all_environments = session.query(Environment).all()
+    for env in all_environments:
+        policies = ResourcePolicyService.find_resource_policies(
+            session=session,
+            group=env.SamlGroupName,
+            resource_uri=env.environmentUri,
+            resource_type=Environment.__name__,
+            permissions=[ATTACH_METADATA_FORM, CREATE_METADATA_FORM],
+        )
+        for policy in policies:
+            for permission in policy.permissions:
+                session.delete(permission)
+                session.commit()
```

Comment:
this deletes all permissions on the `resource_policy`

When downgrading and re-upgrading my user no longer had other Env permissions like `GET_ENVIRONMENT` and similar - need to add a filter here (likely a bug in the other migration script where used as well)

Reply:
- Latest downgrade has a bug - ResourcePolicyPermission has no field `name`
    -  The following resolves for each instance

```
        for policy in policies:
            for resource_pol_permission in policy.permissions:
                if resource_pol_permission.permission.name in [ATTACH_METADATA_FORM, CREATE_METADATA_FORM]:
                    session.delete(resource_pol_permission)
                    session.commit()
```

Reply:
Oh, stupid me.